sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -22,15 +23,15 @@ namespace Azure { namespace Core {
   private:
     static constexpr size_t UuidSize = 16;
 
-    uint8_t m_uuid[UuidSize];
+    std::array<uint8_t, UuidSize> m_uuid;
```

Comment:
What necessitated changing to `std::array` instead of keeping the standard `T[]`?

Reply:
Because it means that the `operator std::array<>` can return a reference without copying the contents.

As far as I know, you can't express `uint8_t[16]` as a return value and have bounds enforced.
And std::vector or other variants aren't value types.

Reply:
Btw, the storage characteristics of `std::array<uint8_t, 16>` and `uint8_t[16]` are identical: https://godbolt.org/z/eqEovMMKo